src/EFCore.PG/Storage/Internal/Mapping/NpgsqlArrayTypeMapping.cs

```diff
@@ -52,5 +74,18 @@ protected override string GenerateNonNullSqlLiteral(object value)
             sb.Append("[]");
             return sb.ToString();
         }
+
+        protected override void ConfigureParameter(DbParameter parameter)
+        {
+            base.ConfigureParameter(parameter);
+
+            if (parameter is NpgsqlParameter npgsqlParameter)
+            {
+                if (NpgsqlDbType.HasValue)
+                    npgsqlParameter.NpgsqlDbType = NpgsqlDbType.Value;
+            }
+            else
+                throw new InvalidOperationException($"Npgsql-specific type mapping {GetType().Name} being used with non-Npgsql parameter type {parameter.GetType().Name}");
```

Comment:
```csharp
base.ConfigureParameter(parameter);

if (NpgsqlDbType is null)
    return;

if (parameter is NpgsqlParameter npgsqlParameter)
    npgsqlParameter.NpgsqlDbType = NpgsqlDbType.Value;
else
    throw new InvalidOperationException($"Npgsql-specific type mapping {GetType().Name} being used with non-Npgsql parameter type {parameter.GetType().Name}");
```

By the way, there is no need to call `Name` explicitly on a type.

Reply:
The point is that I want to throw if the parameter isn't an NpgsqlParameter - it's an argument check. But I changed the code around to make this clearer/cleaner (and also removed the Name), thanks.